pipeline.wdl

```diff
@@ -112,8 +112,11 @@ workflow pipeline {
                     dockerImages = dockerImages
             }
         }
-        IndexedBamFile bamStructs = {"file": sampleJobs.outputBam, "index": sampleJobs.outputBamIndex}
-        BamAndGender bamGenders = object {file: sampleJobs.outputBam, index: sampleJobs.outputBamIndex, gender: sample.gender }
+
+        File bamFile = select_first([preprocessing.recalibratedBam, sampleJobs.outputBam])
+        File bamIndex = select_first([preprocessing.recalibratedBamIndex, sampleJobs.outputBamIndex])
+        IndexedBamFile bamStructs = {"file": bamFile, "index": bamIndex}
+        BamAndGender bamGenders = object {file: bamFile, index: bamIndex, gender: sample.gender }
```

Comment:
This is incorrect, the preprocessed BAM file will end up getting used for expression quantification. Due to the SplitNCigars step, this will lead to inflated read counts for certain genes.

I believe that only the BAM file in `bamGenders` should be changed, `bamStructs` should remain untouched.

Reply:
:+1: good catch.